js/ui/gallery.js

```diff
@@ -453,15 +453,33 @@ var Gallery = CollectionWidget.inherit({
 
         var duplicateCount = Math.min(this._itemsPerPage(), itemsCount);
 
+        var $items = this._getRealItems();
+        var $container = this._itemContainer();
+
         for(i = 0; i < duplicateCount; i++) {
-            this._renderItem(0, items[i]).addClass(GALLERY_LOOP_ITEM_CLASS);
+            this._cloneItemForDuplicate($items[i], $container);
         }
 
         for(i = 0; i < duplicateCount; i++) {
-            this._renderItem(0, items[lastItemIndex - i]).addClass(GALLERY_LOOP_ITEM_CLASS);
+            this._cloneItemForDuplicate($items[lastItemIndex - i], $container);
+        }
+    },
+
+    _cloneItemForDuplicate: function(item, $container) {
```

Comment:
```suggestion
    _cloneItem: function(item, $container) {
```

Reply:
I think the "_cloneItem" name does not show some function specifics like adding of GALLERY_LOOP_ITEM_CLASS